core/forkchoice.go

```diff
@@ -119,13 +128,39 @@ func (f *ForkChoice) ReorgNeeded(current *types.Header, header *types.Header) (b
 			reorg = !currentPreserve && (externPreserve || f.rand.Float64() < 0.5)
 		}
 	}
-	if reorg && types.QuaiNetworkContext != params.PRIME {
-		domReorg, err := f.chain.DomReorgNeeded(header)
-		if err != nil {
-			return false, err
-		}
-		reorg = domReorg
-	}
+
+	// if reorg && types.QuaiNetworkContext != params.PRIME {
+	// 	domReorg, err := f.chain.DomReorgNeeded(header)
+	// 	fmt.Println("domReorg", err)
+	// 	if err != nil {
+	// 		return false, err
+	// 	}
+	// 	reorg = domReorg
+	// }
 
 	return reorg, nil
 }
+
+func (f *ForkChoice) UntwistAndTrim(header *types.Header) error {
```

Comment:
It seems like all UntwistAndTrim is doing is calling PCCRC. Why can we not just call PCCRC in insertChain? It also seems like all PCCRC is doing is checking it's pegs and taking an action. How is that different from RollbackToCanonical?